shim/include/kvm_sregs_idxs.h

```diff
@@ -0,0 +1,288 @@
+#define CS_LIMIT_IDX ((uint32_t)16)
+/** @brief index for CS PRESENT register in kvm segment */
+#define CS_PRESENT_IDX ((uint8_t)17)
+/** @brief index for CS DPL register in kvm segment */
+#define CS_DPL_IDX ((uint8_t)18)
+/** @brief index for CS DB register in kvm segment */
+#define CS_DB_IDX ((uint8_t)19)
+/** @brief index for CS S register in kvm segment */
+#define CS_S_IDX ((uint8_t)20)
+/** @brief index for CS L register in kvm segment */
+#define CS_L_IDX ((uint8_t)21)
+/** @brief index for CS G register in kvm segment */
+#define CS_G_IDX ((uint8_t)22)
+/** @brief index for CS AVL register in kvm segment */
+#define CS_AVL_IDX ((uint8_t)23)
+/** @brief index for CS UNUSABLE register in kvm segment */
+#define CS_UNUSABLE_IDX ((uint8_t)24)
+/** @brief index for CS PADDING register in kvm segment */
+#define CS_PADDING_IDX ((uint8_t)25)
+/** @brief index for DS TYPE register in kvm segment */
+#define DS_TYPE_IDX ((uint8_t)26)
+/** @brief index for DS SELECTOR register in kvm segment */
+#define DS_SELECTOR_IDX ((uint16_t)27)
+/** @brief index for DS BASE register in kvm segment */
+#define DS_BASE_IDX ((uint64_t)28)
+/** @brief index for DS LIMIT register in kvm segment */
+#define DS_LIMIT_IDX ((uint32_t)29)
+/** @brief index for DS PRESENT register in kvm segment */
+#define DS_PRESENT_IDX ((uint8_t)30)
+/** @brief index for DS DPL register in kvm segment */
+#define DS_DPL_IDX ((uint8_t)31)
+/** @brief index for DS DB register in kvm segment */
+#define DS_DB_IDX ((uint8_t)32)
+/** @brief index for DS S register in kvm segment */
+#define DS_S_IDX ((uint8_t)33)
+/** @brief index for DS L register in kvm segment */
+#define DS_L_IDX ((uint8_t)34)
+/** @brief index for DS G register in kvm segment */
+#define DS_G_IDX ((uint8_t)35)
+/** @brief index for DS AVL register in kvm segment */
+#define DS_AVL_IDX ((uint8_t)36)
+/** @brief index for DS UNUSABLE register in kvm segment */
+#define DS_UNUSABLE_IDX ((uint8_t)37)
+/** @brief index for DS PADDING register in kvm segment */
+#define DS_PADDING_IDX ((uint8_t)38)
+/** @brief index for FS TYPE register in kvm segment */
+#define FS_TYPE_IDX ((uint8_t)39)
+/** @brief index for FS SELECTOR register in kvm segment */
+#define FS_SELECTOR_IDX ((uint16_t)40)
+/** @brief index for FS BASE register in kvm segment */
+#define FS_BASE_IDX ((uint64_t)41)
+/** @brief index for FS LIMIT register in kvm segment */
+#define FS_LIMIT_IDX ((uint32_t)42)
+/** @brief index for FS PRESENT register in kvm segment */
+#define FS_PRESENT_IDX ((uint8_t)43)
+/** @brief index for FS DPL register in kvm segment */
+#define FS_DPL_IDX ((uint8_t)44)
+/** @brief index for FS DB register in kvm segment */
+#define FS_DB_IDX ((uint8_t)45)
+/** @brief index for FS S register in kvm segment */
+#define FS_S_IDX ((uint8_t)46)
+/** @brief index for FS L register in kvm segment */
+#define FS_L_IDX ((uint8_t)47)
+/** @brief index for FS G register in kvm segment */
+#define FS_G_IDX ((uint8_t)48)
```

Comment:
You can get rid of these indexes as well (at least the individual bit ones, as you will just need an attrib idx